### PR TITLE
[T5, Testst] Add extensive hard-coded integration tests and make sure PT and TF give equal results

### PR DESCRIPTION
**Pull request by patrickvonplaten**
A direct comparison to google's official model seems quite hard - not sure if that's absolutely needed @thomwolf, @craffel

But some integration tests for T5 would be very nice, to be sure that changes in the future will not break T5.
This PR adds hard-coded integration tests, where the input for summarization is copied from Bart's summarization tests and the input for translation is taken from Appendix D of the official [paper](https://arxiv.org/pdf/1910.10683.pdf)

Checking the expected output for PT, one can see that the output looks quite good! 

- [x] Add PyTorch integration tests
- [x] Verify quality (subjectively for the moment)
- [x] Add TF integration tests
- [x] Same output for PT and TF 

UPDATE:
- Found a big bug in TF Beam Search generation (see comment) -> this PR fixes it

### CODE REVIEW

File: src/transformers/modeling_tf_utils.py [hunk 1]
```diff
@@ -1087,14 +1087,13 @@ def _generate_beam_search(
                 for beam_token_rank, (beam_token_id, beam_token_score) in enumerate(
                     zip(next_tokens[batch_idx], next_scores[batch_idx])
                 ):
-
                     # get beam and token IDs
                     beam_id = beam_token_id // vocab_size
                     token_id = beam_token_id % vocab_size
 
                     effective_beam_id = batch_idx * num_beams + beam_id
                     # add to generated hypotheses if end of sentence or last iteration
```
Comment (patrickvonplaten) [inline src/transformers/modeling_tf_utils.py:1095]: That was a rather big bug (should never use `is` for int/float comparisons in Python!!! - was probably me who did this though...). This bug  led to weird outputs in TF beam search. Before this bug, TF beam search never ended before `max_length` and therefore had weird results.

File: src/transformers/modeling_tf_utils.py [hunk 2]
```diff
@@ -1087,14 +1087,13 @@ def _generate_beam_search(
...
-                    if eos_token_id is not None and token_id.numpy() is eos_token_id:
+                    if (eos_token_id is not None) and (token_id.numpy() == eos_token_id):
                         # if beam_token does not belong to top num_beams tokens, it should not be added
```
Comment (patrickvonplaten) [inline src/transformers/modeling_tf_utils.py:1097]: That was a rather big bug (should never use is for int/float comparisons in Python!!! - was probably me who did this though...). This bug led to weird outputs in TF beam search. Before this bug, TF beam search never ended before max_length and therefore had weird results.

**Review by thomwolf — APPROVED** [review]
Good!

File: src/transformers/modeling_tf_utils.py (hunk 2 above)
Comment (julien-c) [reply src/transformers/modeling_tf_utils.py:1097]: Good catch @patrickvonplaten

**Pull request merged**